prov/gni/src/gnix_ep.c

```diff
@@ -49,6 +52,142 @@ static LIST_HEAD(gnix_nic_list);
 static int gnix_ep_close(fid_t fid);
 static int gnix_ep_bind(fid_t fid, struct fid *bfid, uint64_t flags);
 
+
+/*
+ * function arrays to steer send/receive to ep type specific methods
+ */
+
+static ssize_t (* const send_method[])(struct fid_ep *ep,
```

Comment:
This is kind of difficult to read....if I didn't know that it was a function pointer array I'd be pretty confused as to what was going on. Consider doing something like this: 
```C
typedef ssize_t (*send_func_t)(struct fid_ep *ep, const void *buf, size_t len, 
                               void *desc, fi_addr_t dest_addr, void *context);

static const send_func_t const send_method[] = {
    [FI_EP_MSG] = gnix_ep_send_msg,
    [FI_EP_RDM] = gnix_ep_send_rdm
};
```
Just a suggestion...